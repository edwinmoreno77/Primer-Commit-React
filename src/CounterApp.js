import React, { useState } from 'react'
import PropTypes from 'prop-types'



const CounterApp = ({ value }) => {

    const [counter, setCounter] = useState(0);

    //handleAdd
    const handleReset = () => {
        setCounter(value);

    }
    const handleAdd = () => {
        setCounter(counter + 1);//una forma de hacerlo
        // setCounter((c) => c + 1);// otra forma de hacerlo
    }
    const handleSubtract = () => {
        setCounter(counter - 1);

    }

    return (
        <div>
            <div className='div'>
                <h1>Counter App</h1>
            </div>
            <div className='div'>
                <h2> {counter} </h2>
            </div>
            <div className='div'>
                <button onClick={handleAdd}>+1</button>
                <button onClick={handleReset}>Reset</button>
                <button onClick={handleSubtract}>-1</button>
            </div>

        </div>
    )


}

CounterApp.propTypes = {
    value: PropTypes.number
}


CounterApp.defaultProps = {
    value: 0
}


export default CounterApp;