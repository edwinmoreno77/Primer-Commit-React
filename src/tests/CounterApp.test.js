import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import CounterApp from '../CounterApp';

describe('Pruebas en el CounterApp', () => {

    // let wrapper;
    let wrapper = shallow(<CounterApp />);// de esta forma de mantiene al informacion del intelliSense

    beforeEach(() => {

        wrapper = shallow(<CounterApp />);

    })

    test('Debe de mostrar <CounterApp/> correctamente', () => {

        // const wrapper = shallow(<CounterApp />);
        expect(wrapper).toMatchSnapshot();
    });


    test('Debe de mostrar el valor por defecto de 100', () => {

        const wrapper = shallow(<CounterApp value={100} />);

        const counterText = wrapper.find('h2').text().trim();// .trim() es para eliminar los espacios

        expect(counterText).toBe('100');
    });

    test('Debe de funcionar el boton +', () => {

        wrapper.find('button').at(0).simulate('click');
        // console.log(btn1.html());
        const counterText = wrapper.find('h2').text().trim();

        expect(counterText).toBe('1');


    });

    test('Debe de funcionar el boton -', () => {

        const btn1 = wrapper.find('button').at(2).simulate('click');
        // console.log(btn1.html());
        const counterText = wrapper.find('h2').text().trim();

        expect(counterText).toBe('-1');


    });

    test('Debe funcionar el boton reset', () => {

        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(1).simulate('click');

        const counterText = wrapper.find('h2').text().trim();
        // console.log(counterText);

        expect(counterText).toBe('0');
    })
})