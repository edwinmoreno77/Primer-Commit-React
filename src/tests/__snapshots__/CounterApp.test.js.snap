// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el CounterApp Debe de mostrar <CounterApp/> correctamente 1`] = `
<div>
  <div
    className="div"
  >
    <h1>
      Counter App
    </h1>
  </div>
  <div
    className="div"
  >
    <h2>
       
      0
       
    </h2>
  </div>
  <div
    className="div"
  >
    <button
      onClick={[Function]}
    >
      +1
    </button>
    <button
      onClick={[Function]}
    >
      Reset
    </button>
    <button
      onClick={[Function]}
    >
      -1
    </button>
  </div>
</div>
`;
